### DISCUSSION TITLE
How to Enable Text Wrapping for ui.label in NiceGUI?

### DISCUSSION THREAD

**Discussion by Daniel-Fei** [zauberzeug/nicegui#2960] [Q&A]
### Question

I'm using the NiceGUI library for a project, and I've encountered an issue with text wrapping within a ui.label. Here is part of my code:
```py
with ui.expansion(text='Expand One!', group='group').classes('w-full'):
    with ui.row(wrap=True).classes('w-full'):
        with ui.card().classes('no-shadow border-[1px] w-full'):
            content = 'A'*999
            ui.label(content)
```
I want the text in ui.label(content) to wrap automatically when it exceeds the width of the container. However, it appears that the content is still being displayed on a single line. Is there a way to enable text wrapping for a label in NiceGUI? What might I be missing or need to add to my current setup to achieve this?

Thank you for your help!

**Comment by falkoschindler** [comment]
Hi @Daniel-Fei,

In order to break the text _within words_, you can use the Tailwind class ["break-all"](https://tailwindcss.com/docs/word-break):
```py
ui.label(content).classes('break-all')
```

**Reply by Daniel-Fei** [reply]
Hi @falkoschindler,
It works perfectly, thank you!
